packages/smooth_app/lib/pages/search/search_page.dart

```diff
@@ -124,12 +124,6 @@ class _SearchPageState extends State<SearchPage> {
                           child: searchHelper.value!.widget,
                         ),
                       ],
-                      onPopPage: (Route<dynamic> route, dynamic result) {
```

Comment:
I guess we needed that code.
Removed that because of a new 3.24 syntax?

Reply:
`onPopPage` is now deprecated and the new method doesn't return a value.
So we can't do the same thing anymore.

With my tests, it seems there are no regression, but I'm not fully sure (hence the PR in draft)

Reply:
I wouldn't care that much about this feature being dropped.
Looking back at the code, it's when you search for a term, then search further "in the whole world" - i.e. without the country filter. Then if you go back, do you go back to the country results or the initial search input page: that's the purpose of the dropped code.
That shouldn't prevent you from merging this PR. In the worst case scenario there'll be a new issue, for a minor feature.

Reply:
By using a WillPopScope2, we can't get the same behavior.
It's fixed in the latest commit